addons/binding/org.openhab.binding.homie/src/main/java/org/openhab/binding/homie/handler/HomieDeviceHandler.java

```diff
@@ -0,0 +1,421 @@
+        private final ChannelUID channelUID;
+        private final MqttConnection connection;
+        private final String channelCategory;
+        private final TopicParser topicParser;
+        private NodePropertiesList properties;
+
+        public NodeHandler(String nodeId, HomieConfiguration configString, String channelCategory) {
+            this.nodeId = nodeId;
+            this.channelCategory = channelCategory;
+            this.channelId = nodeId;
+            this.channelUID = new ChannelUID(getThing().getUID(), channelId);
+            this.connection = MqttConnection.fromConfiguration(config, this);
+            this.topicParser = new TopicParser(config.getBaseTopic());
+
+        }
+
+        protected NodePropertiesList getProperties() {
+            return properties;
+        }
+
+        public synchronized void init() throws MqttException {
+            connection.listenForNode(getThing().getUID().getId(), nodeId, this);
+        }
+
+        protected void sendCommand(String property, Command command) {
+            try {
+                connection.send(getThing().getUID().getId(), getNodeId(), property, command);
+            } catch (MqttException e) {
+                logger.error("Error sending message", e);
+            }
+        }
+
+        protected final void handleCommand(Command command) {
+            if (command == RefreshType.REFRESH) {
+                try {
+                    connection.listenForNode(getThing().getUID().getId(), nodeId, this);
+                } catch (MqttException e) {
+                    logger.error("Error handling command", e);
+                }
+            } else {
+                onHandleCommand(command);
+            }
+        }
+
+        protected abstract void onHandleCommand(Command command);
+
+        public String getNodeId() {
+            return nodeId;
+        }
+
+        public String getCategory() {
+            return channelCategory;
+        }
+
+        public ChannelUID getChannelUID() {
+            return channelUID;
+        }
+
+        public Channel getChannel() {
+            return getThing().getChannel(channelId);
+        }
+
+        @Override
+        public void messageArrived(String topic, MqttMessage mqttMessage) throws Exception {
+            String msg = mqttMessage.toString();
+            HomieTopic ht = topicParser.parse(topic);
+            onMqttMessage(ht, msg);
+        }
+
+        protected abstract void onMqttMessage(HomieTopic topic, String message);
+
+        public void setProperties(NodePropertiesList properties) {
+            this.properties = properties;
+        }
+
+    }
+
+    private class ESHNodeHandler extends NodeHandler {
+
+        private String unit;
+        private BigDecimal min;
+        private BigDecimal max;
+        private BigDecimal step;
+        private String itemType = "String";
+        private boolean isReadonly;
+        private String description = "";
```

Comment:
You always seem to override this value from the constructor maybe you should not initialize it here.